app/controllers/administrate/application_controller.rb

```diff
@@ -78,6 +78,28 @@ def destroy
       redirect_to action: :index
     end
 
+    def default_sorting_attribute
+      :id
+    end
+
+    def default_sorting_direction
+      :asc
+    end
+
+    def sorting_attribute
+      params.fetch(resource_name, {}).fetch(
+        :order,
+        default_sorting_attribute,
+      )
+    end
+
+    def sorting_direction
+      params.fetch(resource_name, {}).fetch(
+        :direction,
+        default_sorting_direction,
+      )
+    end
+
```

Comment:
I think these methods should all be private. In a controller, only actions should be public. Making these private still allows developers to override them in their controllers.

Reply:
Woah, you just blew my mind a bit. I'm pretty sure every language I've worked with keeps private as private even from subclasses. There are some completely unnecessary hoops I have jumped through… /me adds Ruby access control research to TODO list

Yes, I agree, private is preferred, will push an update

Reply:
Yeah, I had a similar one back when I first learned of this. Can't blame you! Thank you for the change.